__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,305 @@
+        wallet1,
+        wallet2,
+        walletMultisig,
+      ]);
+
+      // Funds for single input/output tests
+      await wallet1.injectFunds(1000, 0);
+      await wallet2.injectFunds(1000, 0);
+      await walletMultisig.injectFunds(1000, 0);
+
+      tokenTx1 = await wallet1.createToken({ amount: 100, name: 'Token wallet1', symbol: 'TKW1' });
+      tokenTx2 = await wallet2.createToken({ amount: 100, name: 'Token wallet2', symbol: 'TKW2' });
+
+      // Awaiting for updated balances to be received by the websocket
+      await TestUtils.pauseForWsUpdate();
+    } catch (err) {
+      TestUtils.logError(err.stack);
+    }
+  });
+
+  afterAll(async () => {
+    global.config.seeds = {};
+    global.config.multisig = {};
+    await wallet1.stop();
+    await wallet2.stop();
+    await walletMultisig.stop();
+  });
+
+  it('should exchange HTR between 2 wallets', async () => {
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({ send_tokens: [{ value: 10 }] })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[HTR P2PKH]: proposal', { body: response.body });
+
+    let { data } = response.body;
+
+    expect(response.body.isComplete).toBe(false);
+
+    // wallet2 updates the proposal
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({ partial_tx: data, receive_tokens: [{ value: 10 }] })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[HTR P2PKH]: update proposal', { body: response.body });
+    expect(response.body.isComplete).toBe(true);
+
+    data = response.body.data;
+
+    // wallet1: sign data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/get-my-signatures')
+      .send({ partial_tx: data })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[HTR P2PKH]: get signatures', { body: response.body });
+    expect(response.body).toEqual({
+      success: true,
+      isComplete: true,
+      signatures: expect.any(String),
+    });
+    const signature = response.body.signatures;
+
+    // wallet2: sign-and-push data
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal/sign-and-push')
+      .send({ partial_tx: data, signatures: [signature] })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[HTR P2PKH]: push', { body: response.body });
+    expect(response.body.success).toBe(true);
+
+    await TestUtils.pauseForWsUpdate();
+  });
+
+  it('should exchange custom tokens between 2 wallets', async () => {
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({ send_tokens: [{ token: tokenTx1.hash, value: 10 }] })
```

Comment:
I would use the second test to test with `outputs` and `inputs` instead of `send_tokens` and `receive_tokens`